velox/exec/WindowPartition.h

```diff
@@ -20,9 +20,15 @@
 
 /// Simple WindowPartition that builds over the RowContainer used for storing
 /// the input rows in the Window Operator. This works completely in-memory.
+/// WindowPartition supports partial window partitioning to facilitate
+/// RowsStreamingWindowBuild, which means that subsequent calculations within
+/// the WindowPartition do not need to wait until the current partition is fully
+/// ready before commencing. Calculations can begin as soon as a portion of the
+/// rows are ready.
 /// TODO: This implementation will be revised for Spill to disk semantics.
 
 namespace facebook::velox::exec {
+
 class WindowPartition {
```

Comment:
@xiaoxmeng : The access patterns are not different as such. WindowPartition in the typical case is complete since it has all the rows for it. Partial is when the WindowParitition is not complete, it will slide to the next rows in the sorted order. So maybe complete and sliding are better names.

So far we only added assertions when the method is valid for partial but not otherwise as the rest of the methods were applicable for both.

I don't anticipate adding any more complexity for WindowParittion types.

But your suggestion is worth deliberating on. I want to give it more thought before pursuing it. Hope that is okay.

Reply:
@aditi-pandit not sure we need a type but it might be better to have some sanity check on API which is not supposed to be called when say 'isPartial_' is set? Thanks!

Reply:
@xiaoxmeng Add `partial_ ` check to corresponding APIs.